### DISCUSSION TITLE
[Suggestion] Add the ability to make a base with oxygen in other planets

### DISCUSSION THREAD

**Discussion by kristibektashi** [terrarium-earth/Ad-Astra#243] [Ideas]
There should be an air pump block that allows you to make a base in other planets where you can breathe even without the space suit.

How it should work is you place an oxygen generator on a different planet and then you extract the oxygen from the generator (using a pipe mod perhaps) and into the pump. The pump then gradually fills the room with breathable air.

The air pump should only work if the base is completely surrounded (i.e. no holes) or underground. If there is any holes in the base, it should just void the liquid oxygen, cause that's how it would work IRL (oxygen escapes through the hole).

The air pump should require quite a bit of energy to function, and depending on the size of the base, might not completely fill up the room with oxygen, so you might need more pumps and/or generators (this is for balancing reasons).

There should also be an oxygen detector item that detects if the area has oxygen or not and how much (percentage), so that you can see if the air pump works before removing your space suit and fix any potential holes in your base.

If a pump cannot completely fill the room with oxygen, a lower percentage should be shown, and if the percentage is low enough there should be random side effects, such as slowness. If the percentage is too low, you should die just like you would on the surface (but perhaps a bit slower than on the surface if it's above 0 percent).

Another suggestion that fits perfectly with the previous one is to add a set of pressurized doors, where if two of them are placed adjacent to each other, they link and one can only open when the other is closed, perhaps with a delay while the pressure is adjusted. This would allow you to access the outside world without losing your precious oxygen. (They should come in 1x2, 2x2, 3x3 and 4x4 variants, and should also work as regular doors for decoration if only one is placed)

Finally, there should be a heat generator so that you won't freeze on cold planets, and a heat stabilizer so that you won't burn on hot planets. Again, both of these blocks should require quite a bit of energy, and should only work when there are no holes in your base, and they also might not be able to cover the entire base if it's big enough (for balancing). This could be accompanied by a temperature detector item, that tells you temperature of your base, so that you can check if something is wrong and fix it.

The heat generator could also be used to create an environment in which piglins, hoglins etc don't zombify (including mobs added by Ad Astra), by setting the temperature to a high number (something like 200°C, temperature for humans should be around 15-20°C and should be set by default). This could work on any planet and dimension, even the overworld (earth)

**Comment by AlexNijjar** [comment]
uh, everything you’ve suggested already exists in the mod. oxygen distributor for distributing oxygen to a room, 3x3 airlocks, oxygen sensor; it all exists in the mod already. and we have an unreleased version we are working on that added a temperature regulator too

**Reply by kristibektashi** [reply]
> uh, everything you’ve suggested already exists in the mod. oxygen distributor for distributing oxygen to a room, 3x3 airlocks, oxygen sensor; it all exists in the mod already. and we have an unreleased version we are working on that added a temperature regulator too

Sorry my bad, I didn't realize it already existed
